### PR TITLE
[NavigationBar] Implement a semantic color scheme color themer API.

### PR DESCRIPTION
**Pull request by jverkoey**
Also added unit tests and updated a couple examples to make use of the new themer APIs.

Pivotal story: https://www.pivotaltracker.com/story/show/156169509

[Delivers #156169509]

simulator screen shot - iphone x - 2018-04-06 at 11 36 09

### CODE REVIEW

File: components/NavigationBar/src/ColorThemer/MDCNavigationBarColorThemer.h [hunk 1]
```diff
@@ -14,18 +14,31 @@
...
++ (void)applySemanticColorScheme:(nonnull id<MDCColorScheming>)colorScheme
+                 toNavigationBar:(nonnull MDCNavigationBar *)navigationBar;
+
+#pragma mark - Soon to be deprecated
+
 /**
  Applies a color scheme to theme a MDCNavigationBar. Use a UIAppearance proxy to apply a color
  scheme to all instances of MDCNavigationBar.
 
+ This method will soon be deprecated. Consider using +applySemanticColorScheme:toButtonBar: instead.
```
Comment (yarneo) [inline components/NavigationBar/src/ColorThemer/MDCNavigationBarColorThemer.h]: change to "applySemanticColorScheme:toNavigationBar: instead."

File: components/NavigationBar/examples/NavigationBarTypicalUseExample.swift [hunk 2]
```diff
@@ -29,10 +29,12 @@ open class NavigationBarTypicalUseSwiftExample: NavigationBarTypicalUseExample {
     navBar = MDCNavigationBar()
```
Comment (yarneo) [inline components/NavigationBar/examples/NavigationBarTypicalUseExample.swift:29]: maybe not for this PR, but would be nice to get rid of all the forced unwrapping below.
 
```
guard let navBar = navBar else {
  return
}
```

**Review by yarneo — APPROVED** [review]
LGTM % Nits

File: components/NavigationBar/src/ColorThemer/MDCNavigationBarColorThemer.h (hunk 1 above)
Comment (jverkoey) [reply components/NavigationBar/src/ColorThemer/MDCNavigationBarColorThemer.h]: Done.

File: components/NavigationBar/examples/NavigationBarTypicalUseExample.swift (hunk 2 above)
Comment (jverkoey) [reply components/NavigationBar/examples/NavigationBarTypicalUseExample.swift:29]: Agreed.

https://www.pivotaltracker.com/story/show/156619606

**Pull request merged**